### DISCUSSION TITLE
CompilerError: Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON)

### DISCUSSION THREAD

**Discussion by pangpangstudy** [Cyfrin/foundry-full-course-cu#92] [Q&A] [answered]
https://github.com/pangpangstudy/foundry-study-raffle.git

![微信截图_20230616233942](https://github.com/Cyfrin/foundry-full-course-f23/assets/129505105/129a1013-782b-412d-b6a9-14eba315b2a9)

**Comment by pangpangstudy** [comment, marked as answer]
ok, solved. I deleted some unused global variables in utiltest.

**Comment by shedrick4570** [comment]
I started looking for variables to remove as well but then tried using the `--via-ir` tag as mentioned by @inallhonesty here: https://github.com/Cyfrin/foundry-smart-contract-lottery-f23/issues/10.

**Comment by pratikmhatre** [comment]
I ran into same problem and it was very confusing. I just removed some constructor params from Raffle.sol and it worked.

**Reply by sharonphiliplima** [reply]
Hey which ones did you remove exactly?
Were you on lesson 9 too?
this started happening right after `deployKey` 's introduction.
